src/Compilers/CSharp/Test/Semantic/Semantics/RecordStructTests.cs

```diff
@@ -1292,6 +1399,1821 @@ public unsafe record C
                 );
         }
 
-        // PROTOTYPE(record-structs): test `in` and `params` in positional parameters
+        [Fact]
+        public void RecordProperties_01()
+        {
+            var src = @"
+using System;
+record struct C(int X, int Y)
+{
+    int Z = 345;
```

Comment:
>int Z = 345; [](start = 4, length = 12)

Are we testing that this is an error without primary constructor.

Reply:
Yes in `TypeDeclaration_NoInstanceInitializers`

```
        public void TypeDeclaration_NoInstanceInitializers()
        {
            var src = @"
public record struct S
{
    public int field = 42;
    public int Property { get; set; } = 43;
}
";
            // PROTOTYPE(record-structs): this will be allowed in C# 10
            var comp = CreateCompilation(src);
            comp.VerifyDiagnostics(
                // (4,16): error CS0573: 'S': cannot have instance property or field initializers in structs
                //     public int field = 42;
                Diagnostic(ErrorCode.ERR_FieldInitializerInStruct, "field").WithArguments("S").WithLocation(4, 16),
                // (5,16): error CS0573: 'S': cannot have instance property or field initializers in structs
                //     public int Property { get; set; } = 43;
                Diagnostic(ErrorCode.ERR_FieldInitializerInStruct, "Property").WithArguments("S").WithLocation(5, 16)
                );
        }
```

---
In reply to: [590810882](https://github.com/dotnet/roslyn/pull/51720#discussion_r590810882) [](ancestors = 590810882)